lib/foreman_ansible_core/runner/ansible_runner.rb

```diff
@@ -106,6 +108,21 @@ def write_playbook
         File.write(File.join(@root, 'project', 'playbook.yml'), @playbook)
       end
 
+      def write_ssh_key
+        key_path = File.join(@root, 'env', 'ssh_key')
+        File.symlink(ForemanRemoteExecutionCore.settings[:ssh_identity_key_file], key_path)
+
+        if !@passphrase.nil? && !@passphrase.empty?
+          passwords_path = File.join(@root, 'env', 'passwords')
+          secrets = <<~SECRETS
+            ---
+            "for.*/artifacts/.*/ssh_key_data:": "#{@passphrase}"
+          SECRETS
+          File.write(passwords_path, secrets)
```

Comment:
Could we pass the permissions here so there's not even a slightest amount of time during which the file is world readable?

Reply:
In fact this may not even be necessary, the `/tmp/d2154..../` is accesible only for the owner, I just wanted to make sure auditors won't report this as an issue. But yeah, I can add the permissions too.

Reply:
done